test/gtest/uct/test_sockaddr.cc

```diff
@@ -1040,6 +1074,34 @@ UCS_TEST_P(test_uct_sockaddr, ep_disconnect_err_codes)
     }
 }
 
+
+// The idea of the test is to check that execution of a user's conn_request_cb
+// could be done simultaneously with uct_listener_destroy() calling from
+// another thread
```

Comment:
The test calls uct_listener_destroy() from the main thread at the same time conn_request_cb is invoked from the async thread, to make sure this is properly handled